frontend/src/app/api.service.ts

```diff
@@ -0,0 +1,25 @@
+import {Injectable} from '@angular/core';
+
+import {HttpClient} from '@angular/common/http';
+import {Observable} from 'rxjs';
+import {environment} from '../environments/environment';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ApiService {
+
+  constructor(private http: HttpClient) {
+  }
+
+  uploadFile(files): Observable<object> {
+    const formData: FormData = new FormData();
+    formData.append('uploaded_data', files[0], files[0].name);
+    return this.http.post(environment.apiRest + 'file', formData);
```

Comment:
Just for my information - can this be abused by uploading some hacky or infinite file to harm the system or is there some protection in the library or the backend?

Reply:
Well the only checks I make are  for the file format and I secure the filename. Other than that - nothing I think